tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -387,7 +387,7 @@ private void OnError(object sender, EventArgs eventArgs)
                 {
                     AddHeaderTagsFromHttpResponse(httpContext, scope);
 
-                    if (exception != null && !is404)
+                    if (exception != null && !is404 && exception is not AppSec.BlockException)
```

Comment:
I'm not sure if this is correct? If we don't call this, you won't set the status code correctly will we? Or is this handled further up?

Reply:
It's done in the previous method ChooseBlockingMethodAndBlock